frontend/src/hooks/useAuth.tsx

```diff
@@ -0,0 +1,58 @@
+import { useState, useEffect, useRef } from "react";
+import Keycloak from "keycloak-js";
+
+const keycloakClient = new Keycloak({
+  url: import.meta.env.VITE_KEYCLOAK_URL,
+  realm: import.meta.env.VITE_KEYCLOAK_REALM,
+  clientId: import.meta.env.VITE_KEYCLOAK_CLIENT_ID,
+});
+
+const useAuth = () => {
+  const isRun = useRef(false);
+  const [isLogin, setIsLogin] = useState(false);
+  const [keycloak, setKeycloak] = useState<Keycloak | null>(null);
+
+  useEffect(() => {
+    // Protection to not run useEffect twice
+    if (isRun.current) {
```

Comment:
Why is this needed? The empty array as a second param of useEffect should do this no?

Reply:
> The empty array as a second param of useEffect should do this no?

Well, yes and no.
Empty array indeed is used in that case.
During development we are running our application in a Strict mode:
![image](https://github.com/ksummarized/ksummarized.com/assets/45317651/ff566780-52b2-463e-a89c-af77a7c4ddc5)
This strict mode is rendering everything twice during development in order to check for any errors (https://stackoverflow.com/questions/61254372/my-react-component-is-rendering-twice-because-of-strict-mode).
So for development we need to explicitly check for it.